src/functionHandlerProvider/providers/HttpServerHandlerProvider.ts

```diff
@@ -102,3 +102,118 @@ export { handler };`;
+            return response_buffer.write
+
+        # Call the WSGI application and build the response
+        app_response = application(environ, start_response)
+        response_body = b''.join(app_response)
+
+        # Return the formatted response based on content type
+        return format_response(response_body, headers_dict, status_code)
+
+    except Exception as e:
+        # Log errors and return a generic 500 response
+        print("Error processing request:", traceback.format_exc())
+        return {
+            "statusCode": 500,
+            "body": "Internal Server Error",
+            "headers": {"Content-Type": "text/plain"},
+            "error": str(e)
+        }
+
+def create_wsgi_environ(event, http_info, path):
+    """
+    Creates a WSGI environment based on the HTTP information received from the event.
+    """
+    return {
+        'REQUEST_METHOD': http_info.get('method', 'GET'),
+        'PATH_INFO': path,
+        'QUERY_STRING': event.get('query', ''),
+        'REMOTE_ADDR': http_info.get('sourceIp', ''),
+        'CONTENT_TYPE': event.get('headers', {}).get('Content-Type', ''),
+        'CONTENT_LENGTH': str(len(event.get('body', ''))),
+        'wsgi.input': BytesIO(event.get('body', '').encode() if isinstance(event.get('body', ''), str) else event['body']),
+        'wsgi.errors': BytesIO(),
+        'wsgi.url_scheme': 'http',
+        # Add HTTP headers
+        **create_wsgi_headers(event.get('headers', {}))
+    }
+
+def create_wsgi_headers(headers):
+    """
+    Transforms the headers from the event into the WSGI required format (prefixing with HTTP_).
+    """
+    wsgi_headers = {}
+    for header, value in headers.items():
+        wsgi_headers[f"HTTP_{header.upper().replace('-', '_')}"] = value
+    return wsgi_headers
+
+def format_response(response_body, headers, status_code):
+    """
+    Formats the response to return, ensuring the content type is handled appropriately.
+    """
+    content_type = headers.get('Content-Type', 'text/html')
+
+    # Depending on the content type, either decode as text or encode as base64
+    if 'text' in content_type or 'json' in content_type:
+        try:
+            body = response_body.decode('utf-8')  # Assume it's text
+        except UnicodeDecodeError:
+            # If it's not valid UTF-8, treat it as binary
+            print("Warning: Response body is not valid UTF-8. Encoding as base64.")
+            body = base64.b64encode(response_body).decode('utf-8')
+    else:
+        # For non-textual (binary) content, encode as base64
+        body = base64.b64encode(response_body).decode('utf-8')
+
+    return {
+        "statusCode": status_code,
+        "body": body,
+        "headers": headers
+    }
+`;
+
+        await writeToFile(outputPath, handlerFileName, handlerContent);
+    }
+
+    // NOT USED
+    async getLocalFunctionWrapperCode(handler: string, entry: string): Promise<string> {
+        return `${handler} ${entry}`;
+    }
```

Comment:
If `not used`, why is it here? :))

If it should be here because it's called for these object types - at least it should be obvious it's not used - return `undefined` or maybe, `false`.

Otherwise - someday, it will be used, regardless the comment :))

Reply:
I don't necessarily have a context, I copied it from node (HttpServerHandlerProvider)

I think definition is needed because it is derived from an interface, but yes, I would return undefined. Is it ok for nodejs too @radu1122 ?